packages/nimble-components/src/table/specs/data-hierarchy-hld.md

```diff
@@ -0,0 +1,199 @@
+
+Error workflow:
+
+1. Same as steps 1-4 above
+2. Error occurs retrieving data for child rows
+3. User calls `collapseRows(...)` on table instance passing parent row id that was previously expanded.
+4. Row is collapsed and "row loading" indicator is no longer displayed.
+5. Client should also provide appropriate error messaging within the application
+
+### Sorting
+
+Tanstack will sort children within each parent row by the same column that the parent is being sorted by. So, it will not be possible to sort children by a different column value than the one its parent is sorted by.
+
+### Selection
+
+Parent rows when selected, regardless of the selection mode that is set on the table, will _not_ select all of the children under that parent. That is, parent rows will behave in the exact same way as leaf rows with respect to selection. Thus, parents will _not_ show an indetermiate selection state if there are some children selected and some are not, as a parent's selection state is only determined by its own selection<sup>1</sup>.
+
+<sup>1</sup>_The "leaf-only" mode mentioned in the 'Future work' section presents a caveat to this, in which parent rows are explicitly configured to not support certain actions, but we still may allow a user to perform selection through._
+
+### ARIA
+
+A parent row would have the same ARIA expectations of any child row, with the additional need to supply the `aria-expanded` attribute when it is expanded.
+
+Things to consider:
+
+-   Putting an appropriate label/title on the progress indicator for the lazy loaded row placeholder.
+-   Expand/collapse button attributes for the parent row. Likely should just match what is done for the group row.
+
+### Future work
+
+#### Leaf-only mode
+
+There are existing SLE grid variants (namely the Tags grid) that provide a type of hierarchical display that doesn't align exactly with either the proposed 'data hierarchy' outlined in this HLD, nor the grouping capability. Instead, while the parent rows may possibly be presented as a normal row (containing values in some subset of columns that the children have values for), they will not be able to be acted upon in the way child rows can, such as allowing an action menu to be defined on them, or being selected individually.
+
+It is expected that this mode will involve creating a new API on the table in order to fully enable, but it should be able to use the other APIs defined in this HLD to support it, and leverage capabilities such as lazy-loading.
+
+Here are a few of the considerations that were made with respect to this mode:
+
+-   grouping should group on leaf items, not parents (could be cumbersome to accomplish with Tanstack)
+-   action menu only on leafs
```

Comment:
I *think* the following issue may be leaf mode specific: https://github.com/ni/nimble/issues/1609
I'm asking them to confirm which table they are working on to see if they would be using hierarchical leaf mode or if they have some other requirement.

Reply:
That's an interesting case because I believe they want an action menu on a different column based on whether or not a given row is a leaf row. So, I don't think leaf mode by itself would solve their problem.

Reply:
I've added an open question with respect to the Steps grid, but also calling it 'Resolved' stating that the Nimble table is not going to add features for what appears to be a sub-optimal UX.